src/visualizer/visualization.py

```diff
@@ -117,33 +112,69 @@ def __cylinder(self, moment: int, oid: ObjectId):
         x_data = radius*np.cos(l) + self.data[str(moment)][oid][0]
         y_data = radius*np.sin(l) + self.data[str(moment)][oid][1]
         return x_data, y_data
+    
+    def __avg_length(self,Owner : ObjectId) -> float:
+        # gives the average between dimensions as the length of arrow
+        # TODO : it is for now , we need to decide for this
+        sum_dimension , num = 0 , 0
+        owner = self.data["Owners"][Owner]
+        if owner is not None:
+            for oid in owner:
+                s = self.data["shapes"][oid]["type"] 
+                if s == "Cube":
+                    sum_dimension += sum(self.data["shapes"][oid]["dimension"][0:2])
+                    num += 1
+                elif s == "Cylinder":
+                        sum_dimension += self.data["shapes"][oid]["dimension"][0]
+                        num += 1
+        else:
+            s = self.data["shapes"][oid]["type"] 
+            if s == "Cube":
+                sum_dimension += sum(self.data["shapes"][oid]["dimension"][0:2])
+                num += 1
+            elif s == "Cylinder":
+                sum_dimension += self.data["shapes"][oid]["dimension"][0]
+                num += 1
+        avg_length = sum_dimension/float(3*num)
+        
+        return avg_length 
+    
+    def __set_color(self) -> None:
+        # set the color between owner object and its dependent objects
+        m , n = 0 , 0
+        for oid in self.data["Owners"]:
+            n += len(self.data["Owners"][oid]) + 1
+            for i in range(m,n):
+                self.lines2d[i].set_color(f"{self.lines2d[n-1].get_color()}")
+            m += n
```

Comment:
You need to add more comments to you code (not only to this part, in general). It's not immediately obvious what some parts of the code are doing.